tests/slenkins/slenkins_control.pm

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2015 SUSE Linux Products GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+use strict;
+use base 'basetest';
+use testapi;
+use lockapi;
+use mmapi;
+use mm_network;
+use ttylogin;
+
+sub run {
+    my $self = shift;
+
+    ttylogin ('4', "root");
+
+    configure_default_gateway;
+    configure_static_ip('10.0.2.1/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    my $ip_num = 15;
+    my $conf = "";
+    open(FH, get_var('CASEDIR') . "/data/slenkins/" . get_var('SLENKINS_NODEFILE'));
+    my $name;
+    my @nodes;
+    while (<FH>) {
+        my ($var, $value) = split /\s+/, $_;
+        if ($var eq 'node') {
+            $name = $value;
+            push @nodes, $name;
+
+            my $ip = "10.0.2.$ip_num";
+            $conf .= "EXTERNAL_IP_" . uc($name) . "=$ip\n";
+
+            $ip_num++;
+        }
+    }
+    close(FH);
+
+
+    print "$conf\n";
+
+    script_output("
+        mkdir /root/.ssh
+        curl -f -v " . autoinst_url . "/data/slenkins/ssh/id_rsa > /root/.ssh/id_rsa
+        chmod 600 /root/.ssh/*
+        chmod 700 /root/.ssh
+
+        zypper -n --no-gpg-checks ar " . get_var('SLENKINS_REPO') . " slenkins
+
+        #FIXME: the extra packages are workarounds for broken deps
+        zypper -n --no-gpg-checks in " . get_var('SLENKINS_CONTROL') . " susetest-python python-twopence slenkins-tests twopence-shell-client slenkins-utils
+    ", 100);
+
+    for my $n (@nodes) {
+        print "waiting for node $n\n";
+        mutex_lock($n);
+    }
+
+    script_output('
+        #FIXME: can we move this to script_output function?
```

Comment:
I meant the next line.

Without this workaround, if the script exits on error,  script_output function waits until timeout, which can be long.